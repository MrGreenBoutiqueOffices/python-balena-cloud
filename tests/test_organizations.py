"""Test the organizations model for Balena Cloud."""

# pylint: disable=too-many-arguments,too-many-positional-arguments
from __future__ import annotations

from typing import TYPE_CHECKING, Any

import pytest
from aresponses import ResponsesMockServer
from syrupy.assertion import SnapshotAssertion

from . import load_fixtures

if TYPE_CHECKING:
    from balena_cloud import BalenaCloud


async def test_get_organizations(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    balena_cloud_client: BalenaCloud,
) -> None:
    """Test the get_organizations method."""
    aresponses.add(
        "api.balena-cloud.com",
        "/v7/organization",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("organizations/all_organizations.json"),
        ),
    )
    organizations = await balena_cloud_client.get_organizations()
    assert organizations == snapshot


@pytest.mark.parametrize(
    ("param_key", "param_value", "endpoint"),
    [
        ("org_id", 1, "/v7/organization(1)"),
        ("org_handle", "test-handle", "/v7/organization(handle='test-handle')"),
    ],
)
async def test_get_organization(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    balena_cloud_client: BalenaCloud,
    param_key: str,
    param_value: Any,
    endpoint: str,
) -> None:
    """Test the get_organization method."""
    aresponses.add(
        "api.balena-cloud.com",
        endpoint,
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("organizations/organization.json"),
        ),
    )
    # Get the organization based on the parameter key
    organization = None
    if param_key == "org_id":
        organization = await balena_cloud_client.get_organization(org_id=param_value)
    elif param_key == "org_handle":
        organization = await balena_cloud_client.get_organization(
            org_handle=param_value
        )
    assert organization == snapshot
