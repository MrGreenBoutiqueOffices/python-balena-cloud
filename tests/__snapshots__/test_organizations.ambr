# serializer version: 1
# name: test_get_organization[org_handle-test-handle-/v7/organization(handle='test-handle')]
  Organization(id=1, name='Organization_1', handle='org_1', company_name='Company 1', website='None', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_frozen=False)
# ---
# name: test_get_organization[org_id-1-/v7/organization(1)]
  Organization(id=1, name='Organization_1', handle='org_1', company_name='Company 1', website='None', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_frozen=False)
# ---
# name: test_get_organizations
  list([
    Organization(id=1, name='Organization_1', handle='org_1', company_name='Company 1', website='None', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_frozen=False),
    Organization(id=2, name='Organization_2', handle='org_2', company_name='Company 2', website='None', created_at=datetime.datetime(2024, 1, 2, 0, 0, tzinfo=datetime.timezone.utc), is_frozen=False),
  ])
# ---
